app/grandchallenge/notifications/views.py

```diff
@@ -89,13 +88,14 @@ class NotificationList(
     paginate_by = 50
 
     def get_queryset(self):
-        return prefetch_nested_generic_foreign_key_objects(
+        # TODO: refactor prefetch function after migration to new model is complete
+        return prefetch_generic_foreign_key_objects(
             super()
             .get_queryset()
             .select_related(
-                "action__actor_content_type",
-                "action__target_content_type",
-                "action__action_object_content_type",
+                "actor_content_type",
+                "target_content_type",
+                "action_object_content_type",
                 "user",
             )
```

Comment:
This means that for the time that we still display notifications through the action field (i.e. the old way), there will again be lots of queries. Only until we transfer all old notifications to the new format though, so this shouldn't be too bad. I had to change this already because for the new notifications, we no longer have nested generic foreign keys, and displaying them would fail (i.e. the prefetch function would throw an error).